tests/irdl/test_operation_definition.py

```diff
@@ -227,6 +230,78 @@ def test_constraint_var_fail_not_satisfy_constraint():
         op.verify()
 
 
+@irdl_op_definition
+class GenericConstraintVarOp(IRDLOperation):
+    name = "test.constraint_var_op"
+
+    T: ClassVar[VarConstraint[IntegerType | IndexType]] = VarConstraint(
+        "T", base(IntegerType) | base(IndexType)
+    )
```

Comment:
In general do we want these constraints to be a visible part of the operation interface? Would it be better to use `_T` instead?

Reply:
What `_T`?

Reply:
Oh yeah I guess they could be, I tried to modify a minimal amount with these changes.